src/System.Text.Json/tests/Serialization/Stream.WriteTests.cs

```diff
@@ -149,6 +150,112 @@ public static async Task WritePrimitivesAsync()
             int i = await JsonSerializer.DeserializeAsync<int>(stream, options);
             Assert.Equal(1, i);
         }
+
+        private class Session
+        {
+            public int Id { get; set; }
+            public string Title { get; set; }
+            public virtual string Abstract { get; set; }
+            public virtual DateTimeOffset? StartTime { get; set; }
+            public virtual DateTimeOffset? EndTime { get; set; }
+            public TimeSpan Duration => EndTime?.Subtract(StartTime ?? EndTime ?? DateTimeOffset.MinValue) ?? TimeSpan.Zero;
+            public int? TrackId { get; set; }
+        }
+
+        private class SessionResponse : Session
+        {
+            public Track Track { get; set; }
+            public List<Speaker> Speakers { get; set; } = new List<Speaker>();
+        }
+
+        private class Track
+        {
+            public int Id { get; set; }
+            public string Name { get; set; }
+        }
+
+        private class Speaker
+        {
+            public int Id { get; set; }
+            public string Name { get; set; }
+            public string Bio { get; set; }
+            public virtual string WebSite { get; set; }
+        }
+
+        [Theory]
+        [InlineData(0)]
```

Comment:
For my own benefit, was there a specific threshold that this particular bug would appear at? Looking at the code fix, it looks more like an issue with nesting rather than payload size.

Reply:
The default threshold is 75% of the default buffer size (16K) so 12K. However it really determines what is returned from the array pool. 

The bug only occurs when there is a level > 0 and the threshold occurs and the buffer is written to the stream. This means simple tests that don't have arrays or nested objects wouldn't encounter this.